### ISSUE TITLE
Add optional AttributionText field to File & Package level. 

### ISSUE THREAD

**Issue by kestewart** [spdx/spdx-spec#28]
Agreement from those on the call,   yes it should be at File & Package level.
This will be an optional field. 
Useful for generating notice files, etc.

Consider does it make sense for snippets? 

Adding “Attributions” to package field to store required attributions (per Oliver)
Alexios: property named FileAttributionText which will hold the text that has to be reproduced.
It can be considered as a combination of information found in properties like FileCopyrightText, LicenseInfoInFile, FileContributor, but it might not simply the sum of these values.
 
The relative part in the spec could be something like:
 
4.xx   File Attribution Text
4.xx.1 Purpose: This field provides a place for the SPDX data creator to record all attributions found in the file that are required to be communicated. These typically include copyright statement(s), license text, and a disclaimer.
4.xx.2 Intent: The intent is to provide the recipient of the SPDX file with all the legally required attributions in the file, therefore complying with the license obligations.
4.xx.3 Cardinality: Optional, one.
4.xx.4 Data Format: free form text that can (and usually will) span multiple lines
4.xx.5 Tag: "FileAttributionText:"
In Tag:value format, the multiple lines are delimited by "<text>" and "</text>".
Example:

```
FileAttributionText: <text>
#   Copyright (C) 2004 Free Software Foundation, Inc.
#   Written by Scott James Remnant, 2004
#
# This file is free software; the Free Software Foundation gives
# unlimited permission to copy and/or distribute it, with or without
# modifications, as long as this notice is preserved.
</text>
4.xx.6 RDF: property fileAttributionText in class spdx:File
Example:
    	<File rdf:about="...">
           	<fileAttributionText>
#   Copyright (C) 2004 Free Software Foundation, Inc.
#   Written by Scott James Remnant, 2004
#
# This file is free software; the Free Software Foundation gives
# unlimited permission to copy and/or distribute it, with or without
# modifications, as long as this notice is preserved.
           	</fileAttributionText>
    	</File>
```

**Comment by wking** [comment]
On Tue, Sep 05, 2017 at 09:31:30AM -0700, Kate Stewart wrote:
> 4.xx.1 Purpose: This field provides a place for the SPDX data
> creator to record all attributions found in the file that are
> required to be communicated.

This can depend on the context.  For example, comparing BSD-3-Clause
and BSD-4-Clause, both require the copyright notice to be preserved in
source copies, but only the BSD-4-Clause requires a:

  This product includes software developed by …

acknowledgement in “All advertising materials mentioning features or
use of this software”.  If we get an AttributionText field, we need to
be clear enough about the expected context to distinguish between (at
least) these two cases.

**Comment by kestewart** [comment]
Fixed by #208
